engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnJobBehavior.java

```diff
@@ -84,27 +85,38 @@ public BpmnJobBehavior(
     final var type = element.getJobWorkerProperties().getType();
     final var retries = element.getJobWorkerProperties().getRetries();
     final var assignee = element.getJobWorkerProperties().getAssignee();
+    final var candidateGroups = element.getJobWorkerProperties().getCandidateGroups();
```

Comment:
:wrench: 

`element.getJobWorkerProperties()` could be called only once

Another thought is the interface/name of `evalAssigneeExpression` and `evalCandidateGroupExpression`. Instead of passing in an `Expression` you could pass in the `JobWorkerProperties`. Then the whole logic of which fields are relevant to read out and the evaluation of the expression would be captured in this method. 

Another option would be to inline the methods, e.g. `Optional.ofNullable(assignee).map(assignee -> expressionBehavior.evaluateStringExpression(assignee, scopeKey)).orElse(null)`

The way it is now, I feel it is a bit of a mix of abstractions. There is a method with a highly specific name and purpose but with a low level interface based on expressions.

Reply:
An interesting idea, I've played around with it a little and I went for the passing in the `JobWorkerProperties` option. It seems to focus the abstraction level of the `evaluateJobExpressions` method.